generator/transit_generator.cpp

```diff
@@ -111,17 +129,71 @@ struct IsSortedVisitor
   bool m_isSorted = true;
 };
 
-/// \returns ref to a stop at |stops| by |stopId|.
-Stop const & FindStopById(vector<Stop> const & stops, StopId stopId)
+template <typename T>
+void Append(vector<T> const & src, vector<T> & dst)
+{
+  dst.insert(dst.end(), src.begin(), src.end());
+}
+
+void LoadBorders(string const & dir, TCountryId const & countryId, vector<m2::RegionD> & borders)
+{
+  string const polyFile = my::JoinPath(dir, BORDERS_DIR, countryId + BORDERS_EXTENSION);
+  borders.clear();
+  osm::LoadBorders(polyFile, borders);
+}
+
+bool HasStop(vector<Stop> const & stops, StopId stopId)
+{
+  return binary_search(stops.cbegin(), stops.cend(), Stop(stopId));
+}
+
+/// \brief Removes from |items| all items which stop ids is not contained in |stops|.
+/// \note This method keeps relative order of |items|.
+template <class Item>
+void ClipItemsByStops(vector<Stop> const & stops, vector<Item> & items)
+{
+  CHECK(is_sorted(stops.cbegin(), stops.cend()), ());
+
+  vector<Item> itemsToFill;
+  for (auto const & item : items)
+  {
+    for (auto const stopId : item.GetStopIds())
+    {
+      if (HasStop(stops, stopId))
+      {
+        itemsToFill.push_back(item);
+        break;
+      }
+    }
+  }
+
+  items.swap(itemsToFill);
+}
+
+/// \returns ref to an item at |items| by |id|.
+/// \note |items| must be sorted before a call of this method.
+template <class Id, class Item>
+Item const & FindById(vector<Item> const & items, Id id)
 {
-  auto s1Id = equal_range(stops.cbegin(), stops.cend(),
-                          Stop(stopId, kInvalidOsmId, kInvalidFeatureId, kInvalidTransferId,
-                               {} /* line ids */, {} /* point */, {} /* title anchors */));
-  CHECK(s1Id.first != stops.cend(), ("No stop with id:", stopId, "in stops:", stops));
-  CHECK_EQUAL(distance(s1Id.first, s1Id.second), 1, ("A stop with id:", stopId, "is not unique in stops:", stops));
+  auto const s1Id = equal_range(items.cbegin(), items.cend(), Item(id));
+  CHECK_EQUAL(distance(s1Id.first, s1Id.second), 1,
+              ("An item with id:", id, "is not unique or there's not such item. items:", items));
```

Comment:
It would be useful to trace distance(first, second) here too.

Reply:
But the distance(first, second) will be printed any way if the CHECK_EQUAL stops generator.

Reply:
Ah, right! Thanks!